### ISSUE TITLE
[t] Upgrade `django-rest-framework` to latest release 

### ISSUE THREAD

**Issue by FroggyFlox** [rockstor/rockstor-core#2809]
On March 15th, 2024, the Django Rest Framework published their latest release. We should consider upgrading our dependencies accordingly:

https://www.django-rest-framework.org/community/release-notes/#315x-series

**Comment by phillxnet** [comment]
Adding for context our last update of this library:
"Update Django-rest-framework to latest #2738" #2743

**Comment by phillxnet** [comment]
Current pyproject.toml pinning: djangorestframework = "*"

**Comment by phillxnet** [comment]
```
rleap15-5:/opt/rockstor # poetry update
Updating dependencies
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-aResolving dependencies... (33.1s)

Package operations: 3 installs, 28 updates, 3 removals

  • Removing deprecated (1.2.14)
  • Removing pytz (2023.3.post1)
  • Removing wrapt (1.16.0)
  • Updating pycparser (2.21 -> 2.22)
  • Installing backports-tarfile (1.1.0)
  • Updating cryptography (41.0.7 -> 42.0.5)
  • Updating more-itertools (10.1.0 -> 10.2.0)
  • Updating zipp (3.17.0 -> 3.18.1)
  • Updating asgiref (3.7.2 -> 3.8.1)
  • Updating certifi (2023.11.17 -> 2024.2.2)
  • Updating idna (3.6 -> 3.7)
  • Updating importlib-metadata (6.8.0 -> 7.1.0)
  • Updating jaraco-classes (3.3.0 -> 3.4.0)
  • Installing jaraco-context (5.3.0)
  • Installing jaraco-functools (4.0.0)
  • Updating setuptools (69.0.2 -> 69.5.1)
  • Updating sqlparse (0.4.4 -> 0.5.0)
  • Updating typing-extensions (4.8.0 -> 4.11.0)
  • Updating urllib3 (2.1.0 -> 2.2.1)
  • Updating bidict (0.22.1 -> 0.23.1)
  • Updating django (4.2.7 -> 4.2.11)
  • Updating greenlet (3.0.1 -> 3.0.3)
  • Updating jwcrypto (1.5.0 -> 1.5.6)
  • Updating keyring (23.13.1 -> 25.1.0)
  • Updating packaging (23.2 -> 24.0)
  • Updating zope-interface (6.1 -> 6.3)
  • Updating distro (1.8.0 -> 1.9.0)
  • Updating django-pipeline (2.1.0 -> 3.0.0)
  • Updating djangorestframework (3.14.0 -> 3.15.1)
  • Updating gevent (23.9.1 -> 24.2.1)
  • Updating gunicorn (21.2.0 -> 22.0.0)
  • Updating keyring-pass (0.8.1 -> 0.9.3)
  • Updating psycopg (3.1.13 -> 3.1.18)
  • Updating pyzmq (25.1.1 -> 26.0.0)

Writing lock file

```

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2836
